pkg/merger/merger.go

```diff
@@ -91,31 +92,55 @@ type mergeClient interface {
 func MergeAndUpdate(ctx context.Context, client mergeClient, list MergeList, skipValidate, confirm bool) error {
 	ctx, cancel := context.WithCancel(ctx)
 	defer cancel()
-	// Deserialize each proto
-	// TODO: Reading and validating can be done in parallel with a wait group.
+
 	// TODO: Cache the version for each source. Only read if they've changed.
-	shards := map[string]*configpb.Configuration{}
-	// paths, targetPath
+	type Shard struct {
+		name string
+		cfg  *configpb.Configuration
+		err  error
+	}
+
+	shardChan := make(chan Shard)
+	var wg sync.WaitGroup
 
 	for _, source := range list.Sources {
 		if source.Path == nil {
 			return fmt.Errorf("path at %q is nil", source.Name)
 		}
-		cfg, err := config.ReadGCS(ctx, client, *source.Path)
-		if err != nil {
-			return fmt.Errorf("can't read config %q at %s: %w", source.Name, source.Path, err)
-		}
-		if !skipValidate {
-			if err := config.Validate(cfg); err != nil {
-				logrus.WithError(err).WithFields(logrus.Fields{
-					"component":   "config-merger",
-					"config-path": source.Location,
-					"contact":     source.Contact,
-				}).Errorf("config %q is invalid; skipping config", source.Name)
-				continue
+
+		wg.Add(1)
+		go func(src Source) {
```

Comment:
another solution to this problem (which tends to be a little more popular) is the following:

```golang
for _, source := range list.Sources {
  source := source
  go func() {
    // the shadowed source will not change
  }
}
```

`src := source` personally I would recommend `source := source` as this has the advantage of preventing someone from accessing the `source` from `for _, source := range list.Sources`. Right now I could change `src` to `source` on L123 and the compiler will be happy but we'll start using the wrong location

Reply:
I'm not sure if I prefer `source := source` or `go func (source Source)` for shadowing. I went with your recommendation.